service/history/workflow/mutable_state_impl.go

```diff
@@ -1282,50 +1297,18 @@ func (e *MutableStateImpl) addWorkflowExecutionStartedEventForContinueAsNew(
 		req.WorkflowExecutionExpirationTime = &workflowTimeoutTime
 	}
 
-	// History event only has namespace so namespaceID has to be passed in explicitly to update the mutable state
-	var parentNamespaceID string
-	if parentExecutionInfo != nil {
-		parentNamespaceID = parentExecutionInfo.GetNamespaceId()
-	}
-
-	event := e.hBuilder.AddWorkflowExecutionStartedEvent(
-		*e.executionInfo.StartTime,
+	event, err := e.AddWorkflowExecutionStartedEventWithOptions(
+		execution,
 		req,
+		parentExecutionInfo.GetNamespaceId(),
```

Comment:
could parentExecutionInfo be nil? https://github.com/temporalio/temporal/blob/9e91296ed952433bb0b03d1d1d570110d3ae86ad/service/history/workflow/mutable_state_impl.go#L2969